### ISSUE TITLE
Why is az webapp up detecting my runtime when I have specified it vis --runtime?

### ISSUE THREAD

**Issue by charris-msft** [Azure/azure-cli#22117]
## I'm trying to create a new webapp using `az webapp up` and I have specified the runtime with `--runtime "PYTHON:3.9"`, but I get an auto-detect error

**Command Name**
`az webapp up`

**Errors:**
```
Could not auto-detect the runtime stack of your app.
HINT: Are you in the right folder?
For more information, see 'https://go.microsoft.com/fwlink/?linkid=2109470'

```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- `az webapp up -n myWebAppName -g myResourceGroupName --runtime "PYTHON:3.9" -l westus3 --logs -p myPlanName --sku B2`

## Expected Behavior
I expect the command to create the RG, ASP, Web App, and then deploy the app
## Environment Summary
```
Windows-10-10.0.19044-SP0
Python 3.10.3
Installer: MSI

azure-cli 2.35.0

Extensions:
containerapp 0.3.0

Dependencies:
msal 1.17.0
azure-mgmt-resource 20.0.0
```

**Comment by SaurabhSharma-MSFT** [comment]
@charris-msft We are looking into it and get back to you for any additional information.

**Comment by yonzhan** [comment]
route to CXP team

**Comment by SatishBoddu-MSFT** [comment]
Hello @charris-msft 

I was able to reproduce the issue when i was running the command from wrong folder path!
![image](https://user-images.githubusercontent.com/62923178/164121351-3c23a88c-e635-43d2-a807-e817f6efb421.png)

When i change my working path to the right folder then I was able to run the command ` az webapp up` and it detected my Python code runtime automatically!

![image](https://user-images.githubusercontent.com/62923178/164121593-b9176c13-12cd-4bab-90fd-5321166760a6.png)

**Comment by charris-msft** [comment]
Hi @SatishBoddu-MSFT - 
I saw that in the error message and considered it, but I am in the folder that I want it to use. 

Should it even be trying to auto-detect the runtime when I specified the runtime in the arguments?

**Comment by SatishBoddu-MSFT** [comment]
Hello @charris-msft thanks for the response, we are routing this to the service teams' attention for further action!

**Comment by panchagnula** [comment]
@StrawnSC my guess the detection here is because there is on OS & this might be a new app, so wr try to detect language for which default OS to use, but we should honor the  --runtime. Can you check if we can make Changes to use --runtime value to set OS as well, in this scenario.

**Comment by StrawnSC** [comment]
> @StrawnSC my guess the detection here is because there is on OS & this might be a new app, so wr try to detect language for which default OS to use, but we should honor the --runtime. Can you check if we can make Changes to use --runtime value to set OS as well, in this scenario.

This appears to be correct. I checked the places where we attempt to detect the language from the directory context (besides if `--runtime` is not provided). If `--os-type` is not specified, we try to pick an OS based on the directory contents. It should be pretty simple to modify this logic to use the `--runtime` language directly, if provided

**Comment by StrawnSC** [comment]
I have a PR in progress that would fix this: #22592.

@charris-msft I'm also curious to know why the language detection failed on your app. Ideally the CLI would recognize it as a python app even if you left off `--runtime`. Do you have a GitHub link to the app (or a zip file) you could DM me on teams?

**Comment by charris-msft** [comment]
@StrawnSC - 
It was just a test I was setting up so I don't have a repo for it.
If I hit it again, I'll create a repo and share it.
